nixos/modules/services/computing/slurm/slurm.nix

```diff
@@ -3,57 +3,62 @@
 with lib;
 
 let
+  settingsFormat = let
+    listSep = ",";
+    allowedTypes = with types; [ bool int float str attrs ];
+
+    entryToString = key: val:
+        if isAttrs val then concatStringsSep "\n" (mapAttrsToList (subKey: subVal: "${key}=${subKey} ${valueToString subVal}") val)
+        else "${key}=${valueToString val}";
+
+    valueToString = val:
+        if isList val then concatStringsSep listSep (map (x: valueToString x) val)
+        else if isBool val then (if val then "YES" else "NO")
+        else if isAttrs val then concatStringsSep " " (mapAttrsToList (k: v: "${k}=${valueToString v}") val)
+        else toString val;
+
+    in {
+      type = with types; let
+        valueType = oneOf ([
+          (listOf valueType)
```

Comment:
Is the slurm config format actually fully recursive? Can you have arbitrarily deeply nested attributes? From a quick glance I think it's not, in which case the type should be adjusted to match

Reply:
Thanks for the feedback. I rewrote the type definition to use  non-nested attributes. The file is indeed rather "flat". I also replaced `attrs` with `attrsOf`.